packages/app/src/cli/models/extensions/specifications/flow_template.ts

```diff
@@ -0,0 +1,85 @@
+// import {createUIExtensionSpecification} from '../ui.js'
+import {BaseSchema} from '../schemas.js'
+import {createExtensionSpecification} from '../specification.js'
+import {loadLocalesConfig} from '../../../utilities/extensions/locales-configuration.js'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {zod} from '@shopify/cli-kit/node/schema'
+import {AbortError} from '@shopify/cli-kit/node/error'
+import {glob} from '@shopify/cli-kit/node/fs'
+import fs from 'fs'
+
+const FlowTemplateExtensionSchema = BaseSchema.extend({
+  name: zod.string(),
+  description: zod.string(),
+  extensions: zod.array(
+    zod.object({
+      handle: zod.string(),
+      type: zod.literal('flow_template'),
+      title: zod.string(),
+      description: zod.string(),
+      categories: zod.array(zod.string()),
+      pre_install_note: zod.string(),
+      post_install_note: zod.string(),
+      require_app: zod.boolean(),
+      visible: zod.boolean(),
+      enabled: zod.boolean(),
+    }),
+  ),
+})
+
+const spec = createExtensionSpecification({
+  identifier: 'flow_template',
+  schema: FlowTemplateExtensionSchema,
+  singleEntryPath: false,
+  appModuleFeatures: (_) => [],
+  deployConfig: async (config, directory) => {
+    // supporting one extension for now in implementation
+    const extension = config.extensions[0]
+    if (!extension) {
+      throw new AbortError(`Missing extension in ${directory} `, 'Make sure you have defined at least one extension.')
+    }
+    if (config.extensions.length > 1) {
+      throw new AbortError(
+        `More than one extension found in ${directory} `,
+        'Make sure you have only one extension defined. This will be expanded to support more in the future.',
+      )
+    }
+    return {
+      handle: extension.handle,
+      type: extension.type,
+      title: extension.title,
+      description: extension.description,
+      categories: extension.categories,
+      pre_install_note: extension.pre_install_note,
+      post_install_note: extension.post_install_note,
+      require_app: extension.require_app,
+      visible: extension.visible,
+      enabled: extension.enabled,
+      localization: await loadLocalesConfig(directory, 'flow_template'),
+      definition: await loadWorkflow(directory),
```

Comment:
So here you're doing a convention based approach but for both the `flow_trigger` and `flow_action` `schema` property we went for configuration. Basically the `schema` property value is the path to the file. We might need some alignment there @ryanische.

The main pro of configuration was that partners could share their schema between multiple extensions and also be able to use the folder structure that met their need (i.e. the schema didn't need to live side-by-side with the extension config file).

Also was wondering:
- Why is the template file expected to be called `.flow`?
- Could `.flow` be the format (e.g. `email-marketing-template.flow`)

Reply:
This is an interesting case - I'm not sure there would be a need to specify a different location for a template file so I can see the advantage of convention here.

Reply:
I decided to make part of the change now and move the .flow files to the `./templates` folder that way we can just accept more in the future